src/asm/lexer.c

```diff
@@ -2235,6 +2203,50 @@ static int yylex_RAW(void)
 			break;
 		}
 	}
+
+finish:
+	/* Trim right whitespace */
+	while (i && isWhitespace(yylval.tzString[i - 1]))
+		i--;
+
+	if (i == sizeof(yylval.tzString)) {
+		i--;
+		warning(WARNING_LONG_STR, "Macro argument too long\n");
```

Comment:
Shouldn't the warning be given before trimming? Otherwise, a truncation right before whitespace (e.g. `/a {555}a/`) would swallow the warning.

Reply:
Only leading and trailing whitespace is truncated. If you do `mac a<x>a` where `<x>` is 555 spaces, `\1` will be 557 chars long and will warn. If you do `mac aa<x>`, `\1` will only be 2 chars long, so shouldn't warn.

Reply:
But it's trailing whitespace *post*-truncation.
```console
$ cat bork.asm
mac: macro
        PRINTLN "<\1>"
endm

        mac a                                                                                                                                                                                                                                                                              b
$ ./rgbasm --version
rgbasm 543bbc9
$ ./rgbasm bork.asm -Weverything
<a>
```

Reply:
You're right. >_< Fixed, with a test case in macro-arguments.asm.